src/common/mClockCommon.h

```diff
@@ -0,0 +1,59 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2021 SK Telecom
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#pragma once
+
+#include "dmclock/src/dmclock_recs.h"
+
+// the following is done to unclobber _ASSERT_H so it returns to the
+// way ceph likes it
+#include "include/ceph_assert.h"
+
+
+namespace ceph {
+
+namespace dmc = crimson::dmclock;
+
+inline void encode(const dmc::ReqParams &rp,
+                   ceph::buffer::list& bl,
+                   uint64_t features=0) {
+  using ceph::encode;
+  encode(rp.delta, bl);
+  encode(rp.rho, bl);
+}
+
+inline void decode(dmc::ReqParams &rp,
+                   ceph::buffer::list::const_iterator& p) {
+  using ceph::decode;
+  decode(rp.delta, p);
+  decode(rp.rho, p);
+}
+
+inline void encode(const dmc::PhaseType &phase,
+                   ceph::buffer::list& bl,
+                   uint64_t features=0) {
+  using ceph::encode;
+  encode(static_cast<uint8_t>(phase), bl);
+}
+
+inline void decode(dmc::PhaseType &phase,
+                   ceph::buffer::list::const_iterator& p) {
+  using ceph::decode;
+  uint8_t int_phase;
+  decode((uint8_t&)int_phase, p);
+  phase = static_cast<dmc::PhaseType>(int_phase);
+}
```

Comment:
We should probably create denc instances for these (see include/denc.h) since they are fixed size.  Is there a reason we need to define these here rather than where the types are actually defined?

Reply:
These implementation choose not to include versioning.  That's probably ok, but was it intentional?

Reply:
Looking into this.

Reply:
On reflection, it's probably not worth versioning these structures.  It would add overhead to MOSDOp[Reply] encoding, and if we really need to we can use the versioning on MOSDOp[Reply].

It would probably be worth replacing these encode/decode implementations with denc instances, though.  Because the structs are defined elsewhere, you'll need to implement the denc overloads and denc_traits<> template directly.

Reply:
@athanatos The encode/decode implementation is now replaced with denc_traits<> template. Please verify.

Reply:
@athanatos Please review the latest changes that incorporates denc_traits when you get a chance. Thanks!

Reply:
@athanatos Can you please review the mClockCommon.h changes? The changes incorporate the denc_traits as per your suggestion.

Reply:
That part LGTM